runelite-client/src/main/java/net/runelite/client/plugins/wikilookup/WikiLookupPlugin.java

```diff
@@ -0,0 +1,182 @@
+	private Client client;
+
+	@Inject
+	private MenuManager menuManager;
+
+	@Inject
+	private WikiLookupInput inputListener;
+
+	@Inject
+	private KeyManager keyManager;
+
+	private boolean hotKeyPressed = false;
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		keyManager.registerKeyListener(inputListener);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		keyManager.unregisterKeyListener(inputListener);
+	}
+
+	@Subscribe
+	public void onMenuObjectClicked(MenuOptionClicked click)
+	{
+		if (click.getMenuOption().equals(WIKI))
+		{
+			String ourLookupName = Text.removeTags(click.getMenuTarget()).replaceAll("\\(.*?\\)", "").replace(' ', '_');
+			LinkBrowser.browse("http://oldschoolrunescape.wikia.com/wiki/" + ourLookupName);
+		}
+	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!hotKeyPressed)
+		{
+			return;
+		}
+
+		String option = Text.removeTags(event.getOption()).toLowerCase();
+
+		if (event.getType() == MenuAction.EXAMINE_ITEM_GROUND.getId()
+			|| event.getType() == MenuAction.EXAMINE_ITEM.getId()
+			|| event.getType() == MenuAction.EXAMINE_ITEM_BANK_EQ.getId()
+			|| event.getType() == MenuAction.EXAMINE_OBJECT.getId()
+			|| event.getType() == MenuAction.EXAMINE_NPC.getId()
+			|| event.getType() == MenuAction.WIDGET_DEFAULT.getId()
+			|| event.getType() == MenuAction.WIDGET_TYPE_2.getId())
+		{
+			if ((option.equals("read journal:")
+				|| option.equals("cast")
+				|| option.equals("activate"))
+				|| option.contains("play")
+				|| option.equals("examine"))
+			{
+				addBlankOption("cancel", WIKI, event.getTarget());
+			}
+			else if (option.contains("open"))
+			{
+				addBlankOption("cancel", WIKI, event.getOption().replaceFirst("Open ", "").replaceAll("Journal", "") + "Diary");
+			}
+			else if (option.contains("guide"))
+			{
+				addBlankOption("cancel", WIKI, event.getOption().replaceFirst("View ", "").replaceFirst(" guide", ""));
+			}
+		}
+	}
+
+	private void changeOption(int actionID, String newOption)
+	{
+		MenuEntry[] entries = client.getMenuEntries();
+
+		for (int i = 0; i < entries.length; i++)
+		{
+			if (entries[i].getType() == actionID)
+			{
+				entries[i].setOption(newOption);
+				client.setMenuEntries(entries);
+				return;
+			}
+		}
+	}
+
+	private void changeTarget(int actionID, String newTarget)
+	{
+		MenuEntry[] entries = client.getMenuEntries();
+
+		for (int i = 0; i < entries.length; i++)
+		{
+			if (entries[i].getType() == actionID)
+			{
+				entries[i].setTarget(newTarget);
+				client.setMenuEntries(entries);
+				return;
+			}
+		}
+	}
+
+	private void addBlankOption(String toCopy, String newOption, String newTarget)
+	{
+		MenuEntry[] entries = client.getMenuEntries();
+		MenuEntry[] newEntries = new MenuEntry[entries.length + 1];
+		for (int i = 0; i < entries.length; ++i)
+		{
+			if (Text.removeTags(entries[i].getOption()).toLowerCase().contains(toCopy))
+			{
+				MenuEntry toInsert = new MenuEntry();
```

Comment:
What's the reason you dropped the `toBuilder` approach here?

Reply:
I dropped it because there was a conflict with the way the code functions in other places in bytecode. I am not wholly sure but at one point I had changed every instance in the codebase but was getting an error at a certain line but could not edit it. 

This way doesn't change anything else and only is messy in this one spot. Someone with more knowledge should look into making the toBuilder() approach.

Reply:
Alright, clear. It's supposed to be a useful tool, and thus shouldn't be causing annoyances like this. Indeed probably not worth it right now, at least not for this PR.

Reply:
Yeah, I did have it all working. It is really nice, but there was a few places that I could not alter that needed the default constructor method. Honestly I think it was how rs did it? I am not sure.

Reply:
Couldn't you just allow both `MenuEntry.builder()` and `new MenuEntry()`?

Reply:
I have found a way to do just this.